### PR TITLE
Fixing the timestamp passed to the aggregator from the log agent

### PR DESCRIPTION
**Pull request by abhinigam**
Tested it in Denver datacenter.

### CODE REVIEW

File: beater/journalbeat.go [hunk 1]
```diff
@@ -288,6 +288,20 @@ func (jb *Journalbeat) Run(b *beat.Beat) error {
 
 		event["input_type"] = jb.config.DefaultType
 		event["cursor"] = rawEvent.Cursor
+		if tmStr, ok := rawEvent.Fields[timestampField]; ok {
+			tm, err := strconv.ParseInt(tmStr, 10, 64)
+			if err == nil {
+				tmSecs := tm / microseconds
+				tmUSecs := tm % microseconds
```
Comment (connorgorman) [inline beater/journalbeat.go]: inline above?

File: beater/journalbeat.go [hunk 2]
```diff
@@ -288,6 +288,20 @@ func (jb *Journalbeat) Run(b *beat.Beat) error {
...
+				tm := time.Unix(tmSecs, tmUSecs*microsToNanos)
+				event["@timestamp"] = tm.Format("2006-01-02T15:04:05.760738998Z")
+			}
```
Comment (connorgorman) [inline beater/journalbeat.go]: What if err != nil? Is there some action to take?

File: beater/journalbeat.go [hunk 3]
```diff
@@ -288,6 +288,20 @@ func (jb *Journalbeat) Run(b *beat.Beat) error {
...
+		} else {
+			tmSecs := int64(int64(rawEvent.RealtimeTimestamp) / (microseconds))
```
Comment (connorgorman) [inline beater/journalbeat.go]: You do this conversion twice. Maybe make it a function

**Comment by connorgorman** [comment]
@abhinigam Looks good, but can you add a test for at least this use case? It would be great to validate inputs and outputs when we make changes as this logic is getting more complicated

File: beater/journalbeat.go (hunk 2 above)
Comment (abhinigam) [reply beater/journalbeat.go]: In this case logstash aggregator will stamp a time on the server side to the message. This is one of the cases which should never happen based on the data I have looked at so far.

Comment (connorgorman) [reply beater/journalbeat.go]: Maybe just log this case in the off chance it does happen? Could be easier to debug in future

**Comment by abhinigam** [comment]
Connor I have created a case for myself to add tests for journalbeat. 
I will address this in a separate pull request later on.

**Pull request merged**